core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -53,9 +54,7 @@ private[spark] class TaskDescription(
     val name: String,
     val index: Int,    // Index within this task's TaskSet
     val partitionId: Int,
-    val addedFiles: Map[String, Long],
-    val addedJars: Map[String, Long],
-    val addedArchives: Map[String, Long],
+    val artifacts: JobArtifactSet,
```

Comment:
qq, we don't need this change, right? I suspect this is just a part of refactoring? If we want to change this, we should also change `SparkContext.scala` to match to have one higher level `artifact`.

Reply:
> qq, we don't need this change, right? I suspect this is just a part of refactoring?

It's not "needed" per-se but we use `JobArtifactSet` here because we also want to pass in the `uuid` and `replClassDirUri` which are used [here](https://github.com/apache/spark/pull/41625/files/2011b0097a473e173d72e8b916d6386598772d9f#diff-d7a989c491f3cb77cca02c701496a9e2a3443f70af73b0d1ab0899239f3a789dR511-R519) in the `Executor`

> If we want to change this, we should also change SparkContext.scala to match to have one higher level artifact.
To clarify, do you mean replacing [these](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L252-L254) with the `JobArtifactSet`?
I've tried to minimize behaviour/user-visible changes in this PR (it should be a no-op in terms of existing behaviour) since this is a requisite for multi-user Spark Connect sessions but can be later expanded for general multi-user support (i.e outside of Spark Connect) in follow-up PRs.